csrc/scheduler/transpose.cpp

```diff
@@ -587,6 +628,64 @@ std::string getTransposeRuntimeRejectReason(
   }
 #endif
 
+  // TODO: ideally we shouldn't have to manually match schedule transformation
+  // here. It is hard to maintain consistent code logic.
+  if (!scheduler_utils::getViewTVs(fusion).empty()) {
+    const auto index_type = runtime_info.getIndexType();
+    auto params =
+        std::make_shared<TransposeParams>("Transpose heuristics", index_type);
+    maybeBuildVirtualInnerDims(
+        *params,
+        device_multiprocessor_count,
+        n_elems,
+        shape_in_ref1,
+        inner_most_pos1_in_ref1,
+        inner_most_pos2_in_ref1);
+
+    // disallow transpose scheduler when we have a combination of:
+    // 1. view op; and
+    // 2. small transpose transformation
+    // See note [Supporting small transpose dimensions]
+    if (hasSmallTransposeDimensions(params)) {
+      return "Small transpose dimensions and view op cannot be currently be handled by transpose scheduler. See: https://github.com/NVIDIA/Fuser/pull/592";
+    }
+
+    // mimic transform propagation
+    // NOTE: in the actual transpose scheduler, we are applying cacheBefore and
+    // cacheAfter, which I think would mean different propagation is happening
+    // than what's done here. So this might not be bullet proof.
+    TransposeViewPropagator propagator;
+
+    // global schedule traverse dry-run
+    MaxRootDomainInfoSpanningTree entire_dag(reference1);
+    entire_dag.traverse(&propagator);
+
+    if (!propagator.p2c_via_view) {
```

Comment:
hmmm one of the group2/group1 propagation check is killing the transpose scheduler for me on the nanogpt case. I need to take another look at this.